tools/bcl-test-importer/BCLTestImporter/Program.cs

```diff
@@ -0,0 +1,223 @@
+					var types = a.ExportedTypes;
+					if (!types.Any ()) {
+						Console.ForegroundColor = ConsoleColor.DarkYellow;
+						Console.WriteLine ($"WARNING: The following assembly had no types: '{path}'");
+						Console.ForegroundColor = oldColor;
+					}
+					if (verbose)
+						Console.WriteLine ($"Adding to dll {Path.GetFileName (path)} type {types.ElementAt(0).FullName}");
+					dict[Path.GetFileName (path)] = types.First (t => !t.IsAbstract && !t.IsGenericType && t.FullName.Contains ("Test"));
+				} catch (ReflectionTypeLoadException e) {
+					// we did get an exception, possible reason, the type comes from an assebly not loaded, but 
+					// nevertheless we can do something about it, get all the not null types in the exception
+					// and use one of them
+					var types = e.Types.Where (t => t != null).Where (t => !t.IsAbstract && !t.IsGenericType && t.FullName.Contains ("Test"));
+					if (types.Any()) {
+						dict[Path.GetFileName (path)] = types.First ();
+					} else {
+						Console.ForegroundColor = ConsoleColor.DarkYellow;
+						Console.WriteLine ($"WARNING: The following assembly had no types: '{path}'");
+						Console.ForegroundColor = oldColor;
+						continue;
+					}
+				}
+			}
+			return dict;
+		}
+		
+		// return the required assemblies for the given app
+		static IEnumerable<string> GetAssemblyReferences (string assemblyPath, bool verbose = false) {
+			var result = new List<string> ();
+			if (verbose)
+				Console.WriteLine ($"Loading assembly from path {assemblyPath}");
+			var a = Assembly.LoadFile (assemblyPath);
+			return a.GetReferencedAssemblies ().Select ((arg) => arg.Name);
+		}
+
+		public static int Main (string [] args)
+		{
+			var appOptions = new ApplicationOptions ();
+			var options = appOptions.CreateCommandLineOptionSet ();
+			
+			var extra = options.Parse (args);
+			var console = new ConsoleWriter (appOptions);
+			
+			if (extra.Count > 0) {
+				console.WriteWarning ($"The following extra parameters will be ignored: '{ string.Join (",", extra) }'");
+			}
+			// lets validate the options and return any possible error
+			if (!appOptions.OptionsAreValid (out var validationMessage)) {
+				console.WriteError (validationMessage);
+				return 1;
+			}
+
+			if (appOptions.ShouldShowHelp) {
+				options.WriteOptionDescriptions (Console.Out);
+			} else if (!string.IsNullOrWhiteSpace (appOptions.Assembly)) {
+				foreach (var a in GetAssemblyReferences (appOptions.Assembly)) {
+					Console.WriteLine (a);
+				}
+			} else if (appOptions.ShowDict || appOptions.ListAssemblies) {
+				// got the full path, we need to look for diff types of tests, since that changes the glob rule
+				var assemblies = GetTestAssemblies (appOptions.TestsDirectory, appOptions.IsXUnit, appOptions.Verbose);
+				if (assemblies.Count == 0) {
+					Console.WriteLine ("No test assemblies have been found.");
+					return 0;
+				}
+				
+				console.WriteLine ("Using reflection to retrieve a type per assembly.");
```

Comment:
`console` that's pretty confusing when reading the source, since it's intermixed with `Console`
let's find it a new name :)